DQM/L1TMonitor/src/L1TObjectsTiming.cc

```diff
@@ -0,0 +1,480 @@
+        muons_eta_phi.at(index)->Fill(Muon->eta(), Muon->phi());
+      }
+    }
+    
+    // Filling eta-phi map for jets for BX=-2,-1,0,+1,+2
+    for (int itBX = JetBxCollection->getFirstBX(); itBX <= JetBxCollection->getLastBX(); ++itBX) {
+      for (l1t::JetBxCollection::const_iterator jet = JetBxCollection->begin(itBX); jet != JetBxCollection->end(itBX); ++jet) {
+        int index = itBX - std::min(0, 1 - (int)bxrange%2 - (int)std::floor(bxrange/2.)); // the correlation from itBX to respective index of the vector
+        jet_eta_phi.at(index)->Fill(jet->eta(), jet->phi());
+      }
+    }
+
+    // Filling eta-phi map for egamma for BX=-2,-1,0,+1,+2
+    for (int itBX = EGammaBxCollection->getFirstBX(); itBX <= EGammaBxCollection->getLastBX(); ++itBX) {
+      for (l1t::EGammaBxCollection::const_iterator egamma = EGammaBxCollection->begin(itBX); egamma != EGammaBxCollection->end(itBX); ++egamma) {
+        int index = itBX - std::min(0, 1 - (int)bxrange%2 - (int)std::floor(bxrange/2.)); // the correlation from itBX to respective index of the vector
+        egamma_eta_phi.at(index)->Fill(egamma->eta(), egamma->phi());
+      }
+    }
+
+    // Filling eta-phi map for tau for BX=-2,-1,0,+1,+2
+    for (int itBX = TauBxCollection->getFirstBX(); itBX <= TauBxCollection->getLastBX(); ++itBX) {
+      for (l1t::TauBxCollection::const_iterator tau = TauBxCollection->begin(itBX); tau != TauBxCollection->end(itBX); ++tau) {
+        int index = itBX - std::min(0, 1 - (int)bxrange%2 - (int)std::floor(bxrange/2.)); // the correlation from itBX to respective index of the vector
+        tau_eta_phi.at(index)->Fill(tau->eta(), tau->phi());
+      }
+    }
+
+    // Filling eta-phi map for etsum for BX=-2,-1,0,+1,+2
+    for (int itBX = EtSumBxCollection->getFirstBX(); itBX <= EtSumBxCollection->getLastBX(); ++itBX) {
+      for (l1t::EtSumBxCollection::const_iterator EtSum = EtSumBxCollection->begin(itBX); EtSum != EtSumBxCollection->end(itBX); ++EtSum) {
+        int index = itBX - std::min(0, 1 - (int)bxrange%2 - (int)std::floor(bxrange/2.)); // the correlation from itBX to respective index of the vector
+        if (l1t::EtSum::EtSumType::kMissingEt == EtSum->getType()) {
+          etsum_eta_phi_MET.at(index)->Fill(EtSum->phi());
+        }
+        else if (l1t::EtSum::EtSumType::kMissingEtHF == EtSum->getType()) {
+          etsum_eta_phi_METHF.at(index)->Fill(EtSum->phi());
+        }
+        else if(l1t::EtSum::EtSumType::kMissingHt == EtSum->getType()) {
+          etsum_eta_phi_MHT.at(index)->Fill(EtSum->phi());
+        }
+        else if(l1t::EtSum::EtSumType::kMissingHtHF == EtSum->getType()) {
+          etsum_eta_phi_MHTHF.at(index)->Fill(EtSum->phi());
+        }
+      }
+    }
+
+    for(GlobalAlgBlkBxCollection::const_iterator itr = uGtAlgs->begin(0); itr != uGtAlgs->end(0); ++itr) {
+      if(algoBitFirstBxInTrain_ != -1 && algoBitLastBxInTrain_ != -1) {  
```

Comment:
This does not check if the first and last collision in  train algo actually fired for this event. It just checks if the algos exist in the menu. There is an if missing that checks if the two algos fired.